src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -100,39 +104,169 @@
     protected final IGuardBuilding buildingGuards;
 
     /**
-     * Creates the abstract part of the AI.
-     * Always use this constructor!
+     * The interval between sleeping particles
+     */
+    private static final int PARTICLE_INTERVAL = 30;
+
+    /**
+     * The timer for sleeping.
+     */
+    private int sleepTimer = 0;
+
+    /**
+     * The sleeping guard we found
+     */
+    private WeakReference<EntityCitizen> sleepingGuard = new WeakReference<>(null);
+
+    /**
+     * Creates the abstract part of the AI. Always use this constructor!
      *
      * @param job the job to fulfill
      */
     public AbstractEntityAIGuard(@NotNull final J job)
     {
         super(job);
         super.registerTargets(
-          new AITarget(DECIDE, this::decide),
-          new AITarget(GUARD_PATROL, this::patrol),
-          new AITarget(GUARD_FOLLOW, this::follow),
-          new AITarget(GUARD_GUARD, this::guard),
-          new AITarget(GUARD_REGEN, this::regen),
+          new AITarget(DECIDE, this::decide, 20),
+          // This doesnt get checked, decide overrules it! Fix! Tomorrow!
+          new AITarget(GUARD_PATROL, this::shouldSleep, () -> GUARD_SLEEP, 200),
+          new AITarget(GUARD_PATROL, this::decide, 20),
+          new AITarget(GUARD_SLEEP, this::sleep, 1),
+          new AITarget(GUARD_SLEEP, this::sleepParticles, PARTICLE_INTERVAL),
+          new AITarget(GUARD_WAKE, this::wakeUpGuard, 20),
+          new AITarget(GUARD_FOLLOW, this::decide, 20),
+          new AITarget(GUARD_GUARD, this::shouldSleep, () -> GUARD_SLEEP, 200),
+          new AITarget(GUARD_GUARD, this::decide, 20),
+          new AITarget(GUARD_REGEN, this::regen, 40),
           new AITarget(HELP_CITIZEN, this::helping, 20)
-
         );
         buildingGuards = getOwnBuilding();
     }
 
+    int wakeTimer = 0;
+
+    /**
+     * Wake up a nearby sleeping guard
+     *
+     * @return
+     */
+    private IAIState wakeUpGuard()
+    {
+        if (sleepingGuard.get() == null || !(sleepingGuard.get().getCitizenJobHandler().getColonyJob() instanceof AbstractJobGuard) || !sleepingGuard.get()
```

Comment:
Do we have to check here twice, if instanceof guard, and if the job of him is also a guard?